### PR TITLE
GHA: add a workflow to build the pkgdown site on PR

### PR DESCRIPTION
**Pull request by Bai-Li-NOAA**
# What is the feature?

This PR addresses issue #595. It enables FIMS to render the pkgdown site on pull requests.

# How have you implemented the solution?

Thanks, @k-doering-NOAA, for adding a reusable workflow to {ghactions4r}. The workflow call-build-pkgdown.yml has been added to FIMS to use this reusable workflow.

# Does the PR impact any other area of the project?

No
# How to test this change

It has been tested on the feature branch, and the pkgdown site materials have been successfully uploaded on Actions page [here](https://github.com/NOAA-FIMS/FIMS/actions/runs/9505021221)

# Developer pre-PR checklist

- [x] I relied on GitHub actions to :test_tube: things for me while I sat on the :couch_and_lamp:.

### CODE REVIEW

File: .github/workflows/call-build-pkgdown.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Checks that the pkgdown site builds for a repository.
+# this assumes pkgdown is already set up.
+name: call-build-pkgdown
+# on specifies the build triggers. See more info at https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
```
Comment (k-doering-NOAA) [inline .github/workflows/call-build-pkgdown.yml:9]: Just wanted to check: Do we want to only run this on the main branch? I could also see this being helpful for pull requests to main, for instance.

**Review by k-doering-NOAA — CHANGES_REQUESTED** [review]
Hi @Bai-Li-NOAA , sorry for the delay on this! I had one question, but otherwise I think it looks good.

File: .github/workflows/call-build-pkgdown.yml (hunk 1 above)
Comment (kellijohnson-NOAA) [reply .github/workflows/call-build-pkgdown.yml:9]: I was thinking on all PR and a dispatch mode?

Comment (k-doering-NOAA) [reply .github/workflows/call-build-pkgdown.yml:9]: @kellijohnson-NOAA, just to double check, you mean all PRs to main, right? And remove running on push to main?

If so, I think we can modify the build triggers to look like
```yml
on:
  workflow_dispatch:
  pull_request:
    branches:
      - main
```

Comment (kellijohnson-NOAA) [reply .github/workflows/call-build-pkgdown.yml:9]: Yes, that is what I meant because we already have the build and deploy on pushes to main.

Comment (Bai-Li-NOAA) [reply .github/workflows/call-build-pkgdown.yml:9]: Thank you, @k-doering-NOAA and @kellijohnson-NOAA for catching this. The build trigger has been updated.

**Comment by Bai-Li-NOAA** [comment]
Thank you, @k-doering-NOAA. I have rebased the branch.

**Pull request merged**